### ISSUE TITLE
Create training pipeline for SOLOv2 model

### ISSUE THREAD

**Issue by DomMcOyle** [DomMcOyle/TACO-expl#3]
As the title suggests, it is required to implement and test the model an the training pipeline SOLOv2 ([(paper)](https://arxiv.org/pdf/2003.10152.pdf))

**Comment by GiovanniPioDelvecchio** [comment]
![image](https://github.com/DomMcOyle/TACO-expl/assets/54244152/5c27617d-b611-4bbe-9ef7-94de70b3664e)
Not exactly a solvable issue, at least on colab where the torch version is fixed. It turns out that in order to run the solov2 model from the code of the original repository we need to compile some dependencies (using their setup.py) that require torch 1.8, as a feature has been removed in torch 1.11 (see this discussion: https://github.com/open-mmlab/mmdetection3d/issues/1332)

tl:dr;  the folder THC/THC.h is missing from recent pytorch versions, this breaks the solov2 legacy code, thus we have to find an alternative
